server/services/ai.js

```diff
@@ -28,6 +53,179 @@ module.exports = class AIService {
         player.credits = Math.max(0, player.credits);
     }
 
+    async _setupAi(game, player) {
+        player.researchingNext = 'random'; // Set up the AI for random research.
+        
+        // Make sure all stars are marked as not ignored - This is so the AI can bulk upgrade them.
+        const playerStars = this.starService.listStarsOwnedByPlayer(game.galaxy.stars, player._id);
+
+        for (let star of playerStars) {
+            this.starService.resetIgnoreBulkUpgradeStatuses(star);
+        }
+
+        // Clear out any carriers that have looped waypoints.
+        this.carrierService.clearPlayerCarrierWaypointsLooped(game, player);
+
+        const starGraph = this._computeStarGraph(game, player, playerStars);
+
+        // Star systems with computed score based on distance to enemy
+        const starScores = this._computeStarScores(game, player, playerStars, starGraph);
+
+        // Graph of carrier movements for logistics
+        const logisticsGraph = this._createLogisticsGraph(starGraph, starScores);
+        
+        const carrierLoops = this._computeCarrierLoopsFromGraph(logisticsGraph, playerStars);
+
+        const logisticsOrders = this._createCarrierOrders(carrierLoops);
+
+        player.scheduledOrders = logisticsOrders;
```

Comment:
I've made the orders more compact. Recomputing every tick would be way worse (because it would require computing the entire logistics network). Overall, there shouldn't be too many orders usually, as long as the AI has enough credits it won't even store any orders in the database.